namer/namer.cc

```diff
@@ -15,16 +18,1336 @@
+
+    const vector<FoundNameRef> &definitions() const {
+        return _definitions;
+    }
+
+    const vector<FoundClass> &klasses() const {
+        return _klasses;
+    }
+
+    const vector<FoundClassRef> &klassRefs() const {
+        return _klassRefs;
+    }
+
+    const vector<FoundMethod> &methods() const {
+        return _methods;
+    }
+
+    const vector<FoundStaticField> &staticFields() const {
+        return _staticFields;
+    }
+
+    const vector<FoundTypeMember> &typeMembers() const {
+        return _typeMembers;
+    }
+
+    const vector<Modifier> &modifiers() const {
+        return _modifiers;
+    }
+
+    const vector<core::SymbolRef> &symbols() const {
+        return _symbols;
+    }
+
+    friend FoundNameRef;
+};
+
+FoundClassRef &FoundNameRef::klassRef(FoundNames &foundNames) {
+    ENFORCE(kind() == NameKind::ClassRef);
+    ENFORCE(foundNames._klassRefs.size() > idx());
+    return foundNames._klassRefs[idx()];
+}
+const FoundClassRef &FoundNameRef::klassRef(const FoundNames &foundNames) const {
+    ENFORCE(kind() == NameKind::ClassRef);
+    ENFORCE(foundNames._klassRefs.size() > idx());
+    return foundNames._klassRefs[idx()];
+}
+
+FoundClass &FoundNameRef::klass(FoundNames &foundNames) {
+    ENFORCE(kind() == NameKind::Class);
+    ENFORCE(foundNames._klasses.size() > idx());
+    return foundNames._klasses[idx()];
+}
+const FoundClass &FoundNameRef::klass(const FoundNames &foundNames) const {
+    ENFORCE(kind() == NameKind::Class);
+    ENFORCE(foundNames._klasses.size() > idx());
+    return foundNames._klasses[idx()];
+}
+
+FoundMethod &FoundNameRef::method(FoundNames &foundNames) {
+    ENFORCE(kind() == NameKind::Method);
+    ENFORCE(foundNames._methods.size() > idx());
+    return foundNames._methods[idx()];
+}
+const FoundMethod &FoundNameRef::method(const FoundNames &foundNames) const {
+    ENFORCE(kind() == NameKind::Method);
+    ENFORCE(foundNames._methods.size() > idx());
+    return foundNames._methods[idx()];
+}
+
+FoundStaticField &FoundNameRef::staticField(FoundNames &foundNames) {
+    ENFORCE(kind() == NameKind::StaticField);
+    ENFORCE(foundNames._staticFields.size() > idx());
+    return foundNames._staticFields[idx()];
+}
+const FoundStaticField &FoundNameRef::staticField(const FoundNames &foundNames) const {
+    ENFORCE(kind() == NameKind::StaticField);
+    ENFORCE(foundNames._staticFields.size() > idx());
+    return foundNames._staticFields[idx()];
+}
+
+FoundTypeMember &FoundNameRef::typeMember(FoundNames &foundNames) {
+    ENFORCE(kind() == NameKind::TypeMember);
+    ENFORCE(foundNames._typeMembers.size() > idx());
+    return foundNames._typeMembers[idx()];
+}
+const FoundTypeMember &FoundNameRef::typeMember(const FoundNames &foundNames) const {
+    ENFORCE(kind() == NameKind::TypeMember);
+    ENFORCE(foundNames._typeMembers.size() > idx());
+    return foundNames._typeMembers[idx()];
+}
+
+core::SymbolRef FoundNameRef::symbol(const FoundNames &foundNames) const {
```

Comment:
Every other `*Ref` data structure in Sorbet uses the name `.data()` for its corresponding function that does something similar to what these `*Ref` things do. When I saw `name.symbol()` in the code, it wasn't immediately clear to me what it was doing, so I had to read the code.

Can we reuse `data` for the name of this function?

Reply:
I see. Yes, we can do that.

Reply:
Actually, on second thought, that doesn't make sense; a `Symbol` is one kind of found definition (something that is already defined).